### PR TITLE
"What are you in for?" Prisoners can now select their capital offense.

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

New preference, like security department, but for prisoners. They can pick why they're in prison, and they'll get a record about it roundstart.
Classified and Other are for people who want to build their own background, that none of the options currently fit.

Current options:
* Abhorrent Criminal Negligence = Incompetently risked numerous lives.
* Attempted Development of Cloning = Attempted illegal research in the cloning sphere. Cloning, cloning construction, and cloning-related R&D was outlawed in 2560.
* Attempted Murder = Attempted to maliciously kill someone.
* Classified = Consult Legal.
* Corporate Espionage = Conducted espionage against Nanotrasen for commercial purposes.
* Counterfeiting = Engaged in widespread fraud.
* Enemy of the Corporation = Acted as or knowingly aided an enemy of Nanotrasen.
* Grand Sabotage = Engaged in malicious destructive actions seriously threatening Nanotrasen employees and or infrastructure.
* Grand Theft = Stole items of high value or sensitive nature.
* Identity Theft of High-Ranking Figure = Impersonated a high-ranking figure.
* Murder = Maliciously killed someone.
* Mutiny = Attempted to overthrow/subvert Chain of Command.
* Other = Consult Legal.
* Tampering of Artificial Intelligence = Uploaded malicious negligent or otherwise blacklisted directives to an Artificial Intelligence.
* Worship of Blacklisted Deities = Practiced worship of blacklisted deities.

## Why It's Good For The Game

Suggested by https://tgstation13.org/phpBB/viewtopic.php?p=650605#p650605
And I agree with their post:
> It's really dumb whenever I ask them "Well what are you in the prison for?" and they just say some dumb shit like "Oh I just started there.".

This should be a nice little bit of flavor for a mostly roleplay-centric job.

## Changelog

:cl:
add: Prisoners can now select their crime in their character preferences.
/:cl:

### CODE REVIEW

**Comment by ArcaneDefence** [comment]
Where the fuck is tax evasion?

**Comment by CRITAWAKETS** [comment]
I think it needs a few silly ones like "Excessive Jaywalking"

**Comment by tralezab** [comment]
> I think it needs a few silly ones like "Excessive Jaywalking"

I tried to keep it serious but kept the option open for Other, basically

File: code/modules/client/preferences/prisoner_crime.dm [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+/datum/preference/choiced/prisoner_crime/init_possible_values()
+	return GLOB.prisoner_crimes
+
+/datum/preference/choiced/prisoner_crime/apply_to_human(mob/living/carbon/human/target, value)
+	return
+
+/datum/preference/choiced/prisoner_crime/create_default_value()
+	return pick(GLOB.prisoner_crimes)
+
+/datum/preference/choiced/prisoner_crime/is_accessible(datum/preferences/preferences)
```
Comment (Mothblocks) [inline code/modules/client/preferences/prisoner_crime.dm:16]: This should probably check if your highest job is prisoner, just like AI core display: https://github.com/tgstation/tgstation/blob/d91390a4474e511279b6ee7525c5982266b9a858/code/modules/client/preferences/ai_core_display.dm#L18-L22

File: code/modules/client/preferences/prisoner_crime.dm [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+	savefile_key = "prisoner_crime"
+
+/datum/preference/choiced/prisoner_crime/init_possible_values()
+	return GLOB.prisoner_crimes
+
+/datum/preference/choiced/prisoner_crime/apply_to_human(mob/living/carbon/human/target, value)
+	return
+
+/datum/preference/choiced/prisoner_crime/create_default_value()
+	return pick(GLOB.prisoner_crimes)
```
Comment (Mothblocks) [inline code/modules/client/preferences/prisoner_crime.dm:14]: This block shouldn't be necessary, this is the default.

File: code/_globalvars/lists/flavor_misc.dm [hunk 3]
```diff
@@ -192,6 +192,38 @@ GLOBAL_LIST_INIT(security_depts_prefs, sort_list(list(
 	SEC_DEPT_SUPPLY,
 )))
 
+GLOBAL_LIST_INIT(prisoner_crimes, list(
```
Comment (Mothblocks) [inline code/_globalvars/lists/flavor_misc.dm:195]: Doc comment both these lists

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Review by MrMelbert — CHANGES_REQUESTED** [review]

**Comment by CursedBirb** [comment]
Could you add something like Malpractice = Stolen limbs and organs of several crewmembers

**Comment by timothymtorres** [comment]
This is cewl but please add kidnapping and narcotics to the list

**Comment by Farquaar** [comment]
Any way to ask for a random crime so I don't know what I've done until I spawn in?

**Comment by the-orange-cow** [comment]
Consideration: maybe something along the lines of "Biological Terrorism = Created and released a lethal biological agent/lifeform", for all the botanists, virologists, geneticists, xenobiologists, and cytologists out there.

I suppose that grand sabotage/murder covers this, but it has a bit more personalized flair to it.

**Comment by Jezzared** [comment]
Considering we have a "Protective Custody" alt title, i recommend a background to go with it.

**Comment by Mothblocks** [comment]
arm youve just invented a new brain damage json file for everyone to add onto what have you done

**Comment by tralezab** [comment]
> Any way to ask for a random crime so I don't know what I've done until I spawn in?

will add

> arm youve just invented a new brain damage json file for everyone to add onto what have you done

LOOK I DIDNT MEAN TO

**Comment by tralezab** [comment]
> Consideration: maybe something along the lines of "Biological Terrorism = Created and released a lethal biological agent/lifeform", for all the botanists, virologists, geneticists, xenobiologists, and cytologists out there.
> 
> I suppose that grand sabotage/murder covers this, but it has a bit more personalized flair to it.

will add

**Comment by tralezab** [comment]
> Any way to ask for a random crime so I don't know what I've done until I spawn in?

oh and this, will do as well

File: code/modules/jobs/job_types/prisoner.dm [hunk 4]
```diff
@@ -25,6 +25,19 @@
 	rpg_title = "Defeated Miniboss"
 	job_flags = JOB_ANNOUNCE_ARRIVAL | JOB_CREW_MANIFEST | JOB_EQUIP_RANK | JOB_CREW_MEMBER | JOB_NEW_PLAYER_JOINABLE | JOB_ASSIGN_QUIRKS | JOB_CAN_BE_INTERN
 
+/datum/job/prisoner/after_spawn(mob/living/spawned, client/player_client)
+	. = ..()
+	RegisterSignal(GLOB.data_core, COMSIG_MANIFEST_INJECTED(REF(spawned)), .proc/add_pref_crime)
+
+/datum/job/prisoner/proc/add_pref_crime(datum/datacore/source, mob/living/carbon/human/injected_human, list/new_records)
```
Comment (Mothblocks) [inline code/modules/jobs/job_types/prisoner.dm:32]: Needs SIGNAL_HANDLER

**Pull request merged**